src/requirements/Allowlist/AllowlistRequirement.tsx

```diff
@@ -23,7 +23,8 @@ function HiddenAllowlistText({ isEmail }: { isEmail: boolean }) {
 }
 
 const AllowlistRequirement = ({ ...rest }: RequirementProps): JSX.Element => {
-  const requirement = useRequirementContext() as Extract<
+  const requirement = useRequirementContext<"ALLOWLIST" | "ALLOWLIST_EMAIL">()
+  const castedRequirement = requirement as unknown as Extract<
```

Comment:
This might be worth a TODO comment, that we should add `addressCount` to the schema, and remove the cast later on